.github/workflows/build-test.yml

```diff
@@ -1,6 +1,16 @@
 name: build-test
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - "main"
+  pull_request:
+    branches:
+      - "main"
+  schedule:
+    - cron: "0 8 * * *"
```

Comment:
do we want to run this job or do we want a separate job for nightly runs with different requirements? I think a separate job would make sense.